rbd/rbd_test.go

```diff
@@ -629,3 +629,43 @@ func TestTrashImage(t *testing.T) {
 	conn.DeletePool(poolname)
 	conn.Shutdown()
 }
+
+func TestImageMetadata(t *testing.T) {
+	conn, _ := rados.NewConn()
+	conn.ReadDefaultConfigFile()
+	conn.Connect()
+
+	poolname := GetUUID()
+	err := conn.MakePool(poolname)
+	assert.NoError(t, err)
+
+	ioctx, err := conn.OpenIOContext(poolname)
+	require.NoError(t, err)
+
+	name := GetUUID()
+	image, err := rbd.Create(ioctx, name, 1<<22, 22)
+	assert.NoError(t, err)
+
+	err = image.Open()
+	assert.NoError(t, err)
+
+	metadataKey := "mykey"
+	metadataValue := "myvalue"
+
+	err = image.SetMetadata(metadataKey, metadataValue)
```

Comment:
We want to test all error paths as well. Can you add a test to get/set/remove metadata before opening the image?

Reply:
I duplicated all the test for unopen image